src/components/AppSidebar/AppSidebar.vue

```diff
@@ -1,31 +1,308 @@
 <!--
- - @copyright Copyright (c) 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
- -
- - @author Christoph Wurst <christoph@winzerhof-wurst.at>
- -
- - @license GNU AGPL version 3 or any later version
- -
- - This program is free software: you can redistribute it and/or modify
- - it under the terms of the GNU Affero General Public License as
- - published by the Free Software Foundation, either version 3 of the
- - License, or (at your option) any later version.
- -
- - This program is distributed in the hope that it will be useful,
- - but WITHOUT ANY WARRANTY; without even the implied warranty of
- - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- - GNU Affero General Public License for more details.
- -
- - You should have received a copy of the GNU Affero General Public License
- - along with this program. If not, see <http://www.gnu.org/licenses/>.
- -
- -->
+  - @copyright Copyright (c) 2019 John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @author John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
 
 <template>
-	<div id="app-sidebar">
-		<slot />
-	</div>
+	<transition name="slide-right">
+		<aside id="app-sidebar">
+			<header class="app-sidebar-header">
+				<a href="#" class="icon-close" :title="t('core', 'close')"
+					@click="closeSidebar" />
+				<div class="app-sidebar-header__figure" :style="{
```

Comment:
It would be nice if the header is optional: show it only if prop `background` and/or slot `header-background` is given.

Reply:
@korelstar we want to enforce it on nextcloud :)
This is a decision that was made by the ui/ux designer because it actually lighten the sidebar quite a lot, and unify the overall design!

On the note, you could use undraw.co illustrations depending on the note content to show some nice visuals! We talked about it and thought that the way google is doing with calendar (create an event call `dentist` and see a nice header with a toothbrush) is a very nice way of putting a soft touch on helping the user quickly visualise what the data is about, but also is very pleasant to see :wink:

Reply:
Yes, that would be nice for the user.

But in order to implement a dynamic image selection, there are many aspects to consider (how to load images from an external service while keeping privacy, how to find a usefull keyword from the user content, how to ensure that we don't transfer sensitive information to an external service, ...). Will there be a developer guide for all these aspects?

Reply:
Undraw is opensource, so we could create our service to match data to an image while keeping the images offline :)
> unDraw grants you an nonexclusive, worldwide copyright license to download, copy, modify, distribute, perform, and use the assets provided from unDraw for free, including for commercial purposes, without permission from or attributing the creator or unDraw



> Will there be a developer guide for all these aspects

We definitely should yes!

Reply:
But wouldn't that mean that a Nextcloud installation has to ship thousands of (large) images?

Reply:
@korelstar those are svg, so I'd say maybe one hundred images at most? It should not be that big in the end I htink :thinking: 
We should not try to be overwhelming with tons of different images :)

@jancborchardt, @juliushaertl  thoughts?

Reply:
Yes, with those being svg files this shouldn't be a big issue. Also i did some testing and with proper svg optimization, we can reduce the size even further.

I don't think we should enforce using those illustrations, for calendar it makes sense to me, but for notes I don't see a proper use case.

Reply:
@juliushaertl @korelstar I'll tell you what, let's **try** to have illustrations as much as we can (as long as they're relevant, let's not blindly add stuff for the sake of it) but let's **not enforce it**. What do you think? :)

Reply:
I think that this is a good tradeoff. It's a kind of pilot run and if it works, we can enforce it for all, later.